build_runner/lib/src/generate/performance_tracker.dart

```diff
@@ -87,14 +111,45 @@ class BuildPerformanceTracker extends TimeTracker implements BuildPerformance {
   }
 }
 
+/// No-op implementation of [BuildPerformanceTracker].
+///
+/// Read-only fields will throw, and tracking methods just directly invoke their
+/// closures without tracking anything.
+///
+/// Use [BuildPerformanceTracker.noOp] to get an instance.
+class _NoOpBuildPerformanceTracker extends Object
+    with _NoOpTimeTracker
+    implements BuildPerformanceTracker {
+  static final _NoOpBuildPerformanceTracker sharedInstance =
+      new _NoOpBuildPerformanceTracker();
+
+  @override
+  Iterable<BuilderActionTracker> get actions => throw new UnimplementedError();
+
+  @override
+  Iterable<BuildPhaseTracker> get phases => throw new UnimplementedError();
+
+  @override
+  BuilderActionTracker startBuilderAction(
+          AssetId primaryInput, Builder builder) =>
+      new BuilderActionTracker.noOp();
```

Comment:
can we avoid a `new` here?

Reply:
We could make it not a factory, but instead a static getter or something. In reality it isn't a new one though, it is returning a static instance (maybe thats what you don't like though haha).

Thoughts?

Reply:
ah, I didn't read far enough down when I wrote this comment. I'm fine with the `new` since the factory keeps the single instance for us.